docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,304 @@
+This way, even if `spawnTime` is updated through a `MsgUpdateConsumerChain` (see below), we can check on whether the
+new proposed `spawnTime` is less than `issuedTime + (2 months)`. Note that Top N chains can have an arbitrary `spawnTime`
+since their proposals can already be rejected by the community.
+
+For all chains in the prelaunch phase, we keep a mapping between `consumerID` and the underlying `ConsumerAdditionProposal`.
+This way, we can respond to queries that ask for all the consumer chain's parameters. For example, retrieving the
+`spawn_time` of consumer chain with a given `consumerID`.
+
+#### Modify a Consumer Chain
+
+We introduce the `MsgUpdateConsumerChain` message so that the owner of a consumer chain can change its parameters
+(e.g., `spawn_time`, PSS-related parameters, etc.) This message can only be executed by the owner of a consumer chain (see `owner_address` in the `ConsumerChainRecord`).
+
+```protobuf
+message MsgUpdateConsumerChain {
+  // for which chain to change the state ...
+  string consumer_id;
+
+  // if the user intends to change the owner_address
+  // the one issuing this message should be able to sign this message as well?
+  string owner_address;
+  
+  // the chain-id of the new consumer chain
+  string chain_id;
+  // the proposed initial height of new consumer chain.
+  ibc.core.client.v1.Height initial_height;
+
+  ...  (all fields contained in ConsumerAdditionProposal except top_N)
+  
+  uint32 validators_power_cap;
+  uint32 validator_set_cap;
+  repeated string allowlist;
+  repeated string denylist;
+}
+```
+
+The `owner_address` is provided as well and hence a consumer chain can change the current owner account. Because of this,
+the `MsgUpdateConsumerChain` needs to contain two signatures: one for the old and one for the new owner address.
+
+The `initial_height`, `spawnTime`, etc. exist in `MsgUpdateConsumerChain` but can only be modified if the chain
+is still in the prelaunched phase. This can be achieved by looking if there is an ongoing `ConsumerAdditionProposal` for this `consumerID`,
+and then changing the fields of this proposal. The other fields such as `allowlist`, etc. can be changed at any point before or after a chain has started.
+
+#### Stop a Consumer Chain
+With the `MsgStopConsumerChain` we can stop any Opt In chain at any moment. Note that all relevant state for this consumer chain
+remains on the provider's state before getting removed after one unbonding period (of the provider). This is to enable
+potential slashing for any infraction that might have been caused until now. After the unbonding period, the `ConsumerChainRecord`
+associated with this chain is removed. Note however that we never recycle used `consumerID`s. Naturally, this message
+can only be issued by the owner of the consumer chain. Also, any remaining IBC rewards that were to be sent to the provider chain
+are lost.
```

Comment:
> Also, any remaining IBC rewards that were to be sent to the provider chain are lost.

How are IBC rewards lost? If these rewards were sent by the consumer, can be distributed. Otherwise, the IBC rewards will be unescrowed on the consumer.

Reply:
If we stop a consumer chain and delete everything in relation to it then this [call](https://github.com/cosmos/interchain-security/blob/main/x/ccv/provider/keeper/distribution.go#L79) won't iterate through the stopped consumer chain and no tokens would be allocated to the consumer's validators. However, the IBC reward packet could have successfully been sent over to the provider chain (see [here](https://github.com/cosmos/interchain-security/blob/main/x/ccv/provider/ibc_middleware.go#L125)). This seems to be the case today as well.

Reply:
So we should call AllocateTokens before removing a consumer chain, right?

Reply:
Yes, that would be an option. Does not seem critical to me because the sending of IBC rewards over is not synchronized with the provider. In any case, created this [issue](https://github.com/cosmos/interchain-security/issues/2094) because this is orthogonal to permissionless work.